### PR TITLE
KIC 3.0: Annotations + Admission Webhook

### PR DESCRIPTION
**Pull request by mheap**
### Description

Add remaining "How KIC works" docs:

* Annotations
* Admission Webhook

### Testing instructions

Preview link: 

### Checklist 

- [x] Review label added 
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)

### CODE REVIEW

File: app/_src/kubernetes-ingress-controller/concepts/admission-webhook.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+title: Admission Webhook
+type: explanation
+purpose: |
+  Help the user understand. What is the KIC Admission Controller? How do I enable it? What does it validate?
+---
+
+The {{site.kic_product_name}} ships with an admission webhook for KongPlugin
+and KongConsumer resources in the `configuration.konghq.com` API group.
+You can generate TLS certificate and key pair that you need for admission webhook.
+
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/concepts/admission-webhook.md:11]:
```suggestion
{% include_cached /md/kic/prerequisites.md kong_version=page.kong_version disable_gateway_api=false %}
```

**Review by Rajakavitha1 — CHANGES_REQUESTED** [review]
LGTM!!!

**Comment by Rajakavitha1** [comment]
@mheap  I think we need the prereqs to test the gatewayAPI in the late step "Verify Incorrect routes". The only steps that did the work as documented were: https://docs.konghq.com/kubernetes-ingress-controller/latest/deployment/admission-webhook/#verify-incorrect-credential-secrets.
```rajakavitha@Rajakavithas-MacBook-Air ~ % kubectl create secret generic missing-password-credential \
 --from-literal=kongCredType=basic-auth \
 --from-literal=username=foo

secret/missing-password-credential created
rajakavitha@Rajakavithas-MacBook-Air ~ % kubectl create secret generic wrong-cred-credential \
 --from-literal=kongCredType=wrong-auth \
 --from-literal=sdfkey=my-sooper-secret-key

secret/wrong-cred-credential created
```

File: app/_src/kubernetes-ingress-controller/concepts/annotations.md [hunk 2]
```diff
@@ -4,3 +4,16 @@ type: explanation
 purpose: |
   Explain what annotations are used for in KIC. List the top X annotations then link to the reference page
 ---
+
+{{ site.kic_product_name }} uses annotations to add functionality to various Kubernetes resources. Annotations are used when there is not a standardized way to configure the required functionality.
+
+Annotations can be added to a route (`Ingress` or Gateway API resource e.g. `HTTPRoute`), `Service` or `KongConsumer`.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/concepts/annotations.md:10]:
```suggestion
Annotations can be added to a route `Ingress` or Gateway API resource such as `HTTPRoute`, `Service` or `KongConsumer`.
```

File: app/_src/kubernetes-ingress-controller/concepts/annotations.md [hunk 3]
```diff
@@ -4,3 +4,16 @@ type: explanation
...
+
+The most commonly used annotations are:
+
+* [`konghq.com/plugins`](/kubernetes-ingress-controller/{{ page.release }}/reference/annotations/#konghqcomplugins) - Add plugins to a route, service or consumer
+* [`konghq.com/strip-path`](/kubernetes-ingress-controller/{{ page.release }}/reference/annotations/#konghqcomstrip-path) - Strip the path defined in the route and then forward the request to the upstream service
+* [`konghq.com/methods`](/kubernetes-ingress-controller/{{ page.release }}/reference/annotations/#konghqcommethods) - Match specific HTTP methods in the route
+* [`konghq.com/headers.*`](/kubernetes-ingress-controller/{{ page.release }}/reference/annotations/#konghqcomheaders) - Require specific headers in the incoming request to match the defined route
+
+See the [annotations reference](/kubernetes-ingress-controller/{{ page.release }}/reference/annotations/) page for a full list of available annotations and associated documentation.
```
Comment (Rajakavitha1) [inline app/_src/kubernetes-ingress-controller/concepts/annotations.md:19]:
```suggestion
See the [annotations reference](/kubernetes-ingress-controller/{{ page.release }}/reference/annotations/) page for a complete list of available annotations and related documentation.
```

**Review by Rajakavitha1 — CHANGES_REQUESTED** [review]

File: app/_src/kubernetes-ingress-controller/concepts/admission-webhook.md (hunk 1 above)
Comment (mheap) [reply app/_src/kubernetes-ingress-controller/concepts/admission-webhook.md:11]: Adding further down the page

**Comment by Rajakavitha1** [comment]
Hi @mheap . I see this in the preview build. not sure why the pull prereq does not appear.
Screenshot 2023-10-30 at 6 41 50 PM

**Comment by mheap** [comment]
@Rajakavitha1 This is unrelated to the PR and has been fixed on main. I’ll rebase the release branch later

**Comment by Rajakavitha1** [comment]
Hi @mheap ! The verfify incorrect credentials does not work as documented. Error does not appear:
```Error from server: error when creating "STDIN": admission webhook "validations.kong.konghq.com" denied the request: plugin failed schema validation: schema violation (config.foo: unknown field)
rajakavitha@Rajakavithas-MacBook-Air ~ % kubectl create secret generic missing-password-credential \
 --from-literal=kongCredType=basic-auth \
 --from-literal=username=foo

secret/missing-password-credential created
rajakavitha@Rajakavithas-MacBook-Air ~ % kubectl create secret generic wrong-cred-credential \
 --from-literal=kongCredType=wrong-auth \
 --from-literal=sdfkey=my-sooper-secret-key

secret/wrong-cred-credential created
rajakavitha@Rajakavithas-MacBook-Air ~ % ```

**Comment by mheap** [comment]
@Rajakavitha1 This is expected.

In KIC 2.x this will not error unless the credentials are attached to a consumer.
In KIC 3.x there will be an error

**Review by Rajakavitha1 — APPROVED** [review]

**Comment by Rajakavitha1** [comment]
LGTM!!!

**Pull request merged**